### DISCUSSION TITLE
Gettting this error running brownie test -k test_get_entrance_fee --network rinkeby

### DISCUSSION THREAD

**Discussion by avikat** [smartcontractkit/full-blockchain-solidity-course-py#1021] [Q&A] [answered]
Is it neccessary to write test cases always get error in test cases and i dont know how to solve it.
PS C:\Users\AnshulGupta\Desktop\demos\smartcontract_lottery> brownie test -k test_get_entrance_fee --network rinkeby                                                                                   ============================
INFO: Could not find files for the given pattern(s).
Brownie v1.16.4 - Python development framework for Ethereum

======================================================================================== test session starts =========================================================================================
platform win32 -- Python 3.10.2, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: C:\Users\AnshulGupta\Desktop\demos\smartcontract_lottery
plugins: eth-brownie-1.16.4, hypothesis-6.21.6, forked-1.3.0, xdist-1.34.0, web3-5.23.1
collected 1 item
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\_pytest\main.py", line 269, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\_pytest\main.py", line 322, in _main
INTERNALERROR>     config.hook.pytest_collection(session=session)
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\pluggy\_hooks.py", line 265, in __call__
INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\pluggy\_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\pluggy\_callers.py", line 60, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\pluggy\_result.py", line 60, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\pluggy\_callers.py", line 39, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\_pytest\main.py", line 333, in pytest_collection
INTERNALERROR>     session.perform_collect()
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\_pytest\main.py", line 641, in perform_collect
INTERNALERROR>     hook.pytest_collection_finish(session=self)
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\pluggy\_hooks.py", line 265, in __call__
INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\pluggy\_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\pluggy\_callers.py", line 55, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\brownie\test\managers\runner.py", line 258, in pytest_collection_finish
INTERNALERROR>     brownie.network.connect(CONFIG.argv["network"])
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\brownie\network\main.py", line 40, in connect
INTERNALERROR>     web3.connect(host, active.get("timeout", 30))
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\brownie\network\web3.py", line 52, in connect
INTERNALERROR>     uri = _expand_environment_vars(uri)
INTERNALERROR>   File "C:\Users\AnshulGupta\AppData\Local\Programs\Python\Python310\lib\site-packages\brownie\network\web3.py", line 183, in _expand_environment_vars
INTERNALERROR>     raise ValueError(f"Unable to expand environment variable in host setting: '{uri}'")
INTERNALERROR> ValueError: Unable to expand environment variable in host setting: 'https://rinkeby.infura.io/v3/$WEB3_INFURA_PROJECT_ID'

==============================================================

**Comment by avikat** [comment, marked as answer]
it is resolved

**Reply by n4n0b1t3** [reply]
Please describe the solution which solved the problem. This helps other developers with a similar problem.

**Reply by avikat** [reply]
Sorry, it was such a silly mistake. i miss the dotenv: .env in brownie-config.yaml.

**Reply by n4n0b1t3** [reply]
You won't believe how many silly mistakes everyone here makes every day, and your post will definitely help others. Thank you for your contribution :-)
